ui/raidboss/popup-text.ts

```diff
@@ -786,6 +791,9 @@ export class PopupText {
     this.data = this.getDataObject();
     this.StopTimers();
     this.triggerSuppress = {};
+
+    for (const init of this.dataInitializers)
+      this.data = init(this.data);
```

Comment:
I realize that TypeScript will prevent this for built-in triggers, but I think we also need to handle the case of "bad" user triggers clobbering data here.  We should definitely print errors (and reset data again?) if somebody forgets to return anything from `initData` at all.

(Unsure if we should also check if all the fields in what is returned from `this.getDataObject()` are also the same values in this final `this.data`? This only gets called infrequently on wipe, so it might be worth being loud here if somebody holds it wrong.)

Reply:
So, my thought here is that I'd rather store the return in a temp var and check that it has the `me` property or something, rather than clobbering, because that gets a bit odd with having to Object.apply or something?

Reply:
Updated to reflect this. Let me know if this still doesn't make sense.

Reply:
Not that I'm saying we have to clobber, but I don't think that requires `Object.apply`.  You could do something like:

```typescript
    this.data = {
      ...this.data,
      ...this.getDataObject(),
    };
```

Reply:
Alright, how's this latest version look? Sort of a both approach?

Reply:
Yeah, looks good.  Then one init function that forgets to return something doesn't break all the other init functions.

Reply:
Maybe this should log to console as an `else`?

Reply:
Sounds good!